resources/views/auth/login.blade.php

```diff
@@ -9,3 +9,12 @@
     </div>
 </div>
 @endsection
+
+@section('scripts')
```

Comment:
I'm not sure this is the right place for this.  It will only set it if the user visits the login page.  What about when the user remembers their login, then is in a different timezone?

This also requires the route to be unauthenticated.

Take a look at the set_resolution route, it is very similar to what you want to do.  (although it does need to be removed at some point :D)

Reply:
I implemented it this way because there is no trigger for the timezone like there is for the screen resolution.  This means that to make this dynamic I would need to post the current browser timezone on every page load just in case it's changed, and then check to see if the user has hard-coded their timezone.  I don't think the additional load is worthwhile, and would expect people to want to see consistent timezones instead of it changing.  

I'm happy to hear if you have any further thoughts on eliminating the additional requests, or if you still think it's worth making it more dynamic.

I have had a second thought on this feature - should I be aiming to record a device timezone, and having graphs shown in the timezone local to the device?  This could become a global option to enable/disable showing graphs in the device timezone.

Reply:
IMO, everything should default to the browser timezone by default allowing the user to override.

You could definitely write the javascript in such a way that a post is only made when the browser timezone differs from what is given by the server (either the default server time or whatever is stored in the user session).

Reply:
I think this is a very worthwhile thing to solve.  But it is not simple.  Mostly, because time in LibreNMS is a mess :)

Reply:
How does the latest commit look?